apps/data-pipeline/websoc-scraper/src/lib.ts

```diff
@@ -853,6 +846,48 @@ export async function scrapeTerm(
   });
 }
 
+export async function ingestChunk(
+  db: ReturnType<typeof database>,
+  term: Term,
+  lower: `${number}`,
+  upper: `${number}`,
+) {
+  const sectionCodes = `${lower}-${upper}`;
+  console.log(`Scraping chunk ${sectionCodes}`);
+  try {
+    const resp = await request(term, {
+      sectionCodes,
+      cancelledCourses: "Include",
+    }).then(normalizeResponse);
+    if (resp.schools.length) await doChunkUpsert(db, term, resp, null);
+    await sleep(1000);
+  } catch (e) {
+    /*
+     assuming network, etc. conditions are fine, we have more than 900 sections here
+     this means we somehow overran our 1% tolerance
+     that's okay; we can be suboptimal this time so we get all the sections that exist.
+     we're going to recompute the chunks at the start of the next scrape,
+     so that one will run optimally, given no such failure occurs again
+
+     we're going to bisect this chunk and try the two halves separately; eventually,
+     we'll have <= 900 valid sections in a chunk and we'll be in the clear
+    */
+    const lowerInt = baseTenIntOrNull(lower) as number;
```

Comment:
If you're going to always cast to `number` I assume this means you expect `lower` and `upper` to always be a string containing a number. Therefore, you should use `Number.parseInt(x, 10)`.

Also not quite sure why `lower` and `upper` are being passed as `${number}` anyway, but that's addressed above.